system_tests/pubsub.py

```diff
@@ -246,5 +263,12 @@ def test_fetch_delete_subscription_w_deleted_topic(self):
                    if subscription.name == ORPHANED]
         self.assertEqual(len(created), 1)
         orphaned = created[0]
+
+        def _no_topic(instance):
+            return instance.topic is None
+
+        retry = RetryInstanceState(_no_topic)
+        retry(orphaned.reload)()
```

Comment:
I had a hard time grokking `RetryInstanceState` (you might say I found it muddy).

> it won't work if the caller actually needs the real result, too.

Just have `_reload_orphaned` return a tuple of the instance and the result